### PR TITLE
[Issue-295] Add Python code examples in the Flink SQL samples

### LINKED ISSUE #295: Add Python code examples in the Flink SQL samples
**Problem description**
The `pravega-flink-connector-sql-samples` in `scenarios` only supports the java version. Adding a python version would help those from other backgrounds.

**Problem location**
`pravega-flink-connector-sql-samples`

**Suggestions for an improvement**
https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/python/table_api_tutorial.html

### PR DESCRIPTION
**Pull request by thekingofcity**
**Change log description**
The pravega-flink-connector-sql-samples in scenarios only supports the java version. Adding a python version would help those from other backgrounds.

**Purpose of the change**
Fixes #295 

**What the code does**
Add a corresponding python version to the `pravega-flink-connector-sql-samples` in `scenarios`.

**How to verify it**
User could follow the `python.md` and complete the tutorial just as the java version.

### CODE REVIEW

File: scenarios/pravega-flink-connector-sql-samples/src/main/python/common.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    destLocationBorough STRING,
+    destLocationZone STRING,
+    destLocationServiceZone STRING,
+    {watermark_ddl}
+) with (
+    'connector' = 'pravega',
+    'controller-uri' = '{args.controller_uri}',
+    'scope' = '{args.scope}',
+    'scan.execution.type' = 'streaming',
+    'scan.reader-group.name' = '{reader_group_name}',
```
Comment (crazyzhou) [inline scenarios/pravega-flink-connector-sql-samples/src/main/python/common.py:62]: This line forcing the RG is not recommended now, we can remove this.

**Review by crazyzhou — CHANGES_REQUESTED** [review]
I will retest again for a second round review, besides these comments, please change the target branch to `dev` instead of master.

File: scenarios/pravega-flink-connector-sql-samples/python.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+```
+The follwing files are downloaded and used for this sample application
+https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2018-01.csv
+https://s3.amazonaws.com/nyc-tlc/misc/taxi+_zone_lookup.csv
+```
+
+## Pre-requisites
+1. Pravega running (see [here](http://pravega.io/docs/latest/getting-started/) for instructions)
+2. Build [pravega-samples](https://github.com/pravega/pravega-samples) repository
+3. Python version 3.8
```
[the remark this answers is not in the document]
Comment (thekingofcity) [reply scenarios/pravega-flink-connector-sql-samples/python.md:13]: Until now, the `apache-flink` package only supports python from 3.6 to 3.8
https://pypi.org/project/apache-flink/#files

File: scenarios/pravega-flink-connector-sql-samples/python.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+4. (optional) Prepare a virtual environment:
+   `pip install virtualenv && virtualenv pyflink && cd pyflink && source bin/activate`
+5. `pip install apache-flink 1.12.4`
```
[the remark this answers is not in the document]
Comment (thekingofcity) [reply scenarios/pravega-flink-connector-sql-samples/python.md:16]: Do we need to specify the `<flink version>` somewhere else in the document?

File: scenarios/pravega-flink-connector-sql-samples/python.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+6. (optional) Redirect the python you want to use to `python`:
```
[the remark this answers is not in the document]
Comment (thekingofcity) [reply scenarios/pravega-flink-connector-sql-samples/python.md:17]: It is a must to use python at version 3 and have `python` command pointing to it.
See https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/python/installation.html#environment-setup

[the remark this answers is not in the document]
Comment (crazyzhou) [reply scenarios/pravega-flink-connector-sql-samples/python.md:17]: For all the python environment setup, we can directly refer to Flink document: https://ci.apache.org/projects/flink/flink-docs-stable/docs/dev/python/installation/

**Review by crazyzhou — CHANGES_REQUESTED** [review]
The python code needs to have license headers.

File: scenarios/pravega-flink-connector-sql-samples/python.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    The default option assumes that Pravega is running locally. You can override it by passing the controller URI options. The `create-stream` option allows the program to create the scope and stream (enabled by default).
+
+    > bin/tableapi-samples --runApp Prepare
+
+    The above command loads the taxi dataset records to Pravega and prepares the environment for testing.
+
+2. Popular Destination
+
+    > flink run --python ./bin/popular_destination_query.py --pyFiles ./bin/common.py --jarfile ~/pravega-connectors-flink-1.11_2.12-0.9.1.jar
```
Comment (crazyzhou) [inline scenarios/pravega-flink-connector-sql-samples/python.md:41]: We need to generalize the connector version as `--jarfile <path/to/pravega-flink-connector>`

File: scenarios/pravega-flink-connector-sql-samples/python.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2018-01.csv
+https://s3.amazonaws.com/nyc-tlc/misc/taxi+_zone_lookup.csv
+```
+
+## Pre-requisites
+1. Pravega running (see [here](http://pravega.io/docs/latest/getting-started/) for instructions)
+2. Build the [pravega-samples](https://github.com/pravega/pravega-samples) by following the [instructions](https://github.com/pravega/pravega-samples#pravega-samples-build-instructions)
+3. Python and its packages, see [pyflink setup](https://ci.apache.org/projects/flink/flink-docs-stable/docs/dev/python/installation/)
+4. Get the latest Pravega Flink connectors at [the release page](https://github.com/pravega/flink-connectors/releases)
+
```
Comment (crazyzhou) [inline scenarios/pravega-flink-connector-sql-samples/python.md:15]: This also require Flink running, right?

**Review by crazyzhou — CHANGES_REQUESTED** [review]

**Review by crazyzhou — APPROVED** [review]
LGTM, if you can screenshot some of the results in the PR description or comment, it would be better.

**Comment by thekingofcity** [comment]
After executing the `prepare_main` and `popular_taxi_vendor`, the result could be viewed at the Flink UI.

image

**Pull request merged**